### PR TITLE
feat:【数据统计】用户态数据统计- 总览 #6587

### PR DESCRIPTION
**Pull request by carlyin0801**
feat:【数据统计】用户态数据统计- 总览 #6587
#6587

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-ci?pullRequest=7016) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-ci?pullRequest=7016) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: carlyin0801<br/>:white_check_mark: yjieliang<br/>:x: v_yjieliang<br/><hr/>**v_yjieliang** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-ci?pullRequest=7016) it.</sub>

File: src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/constant/Constants.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    const val BK_ERROR_COUNT_SUM = "errorCountSum"
+    const val BK_ERROR_COUNT = "errorCount"
+    const val BK_TOTAL_EXECUTE_COUNT_SUM = "totalExecuteCountSum"
+    const val BK_SUCCESS_EXECUTE_COUNT_SUM = "successExecuteCountSum"
+    const val BK_TOTAL_AVG_COST_TIME_SUM = "totalAvgCostTimeSum"
+
+    const val MAX_CREATE_COUNT = 100
+    const val DEFAULT_LIMIT_NUM = 10
+    const val METRICS_PIPELINE_ID_EXPIRED = 7200L
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/constant/Constants.kt:71]: 最好搞成可配置化，不要写死

File: src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/AtomDisplayConfigManageService.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    /**
+     * 获取项目下可供选择展示的插件
+     * @return 项目下展示插件配置报文
+     */
+    fun getOptionalAtomDisplayConfig(
+        projectId: String,
+        userId: String,
+        keyword: String?,
+        page: Int?,
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/AtomDisplayConfigManageService.kt:70]: 这个不能分页吗，强制分页吧

File: src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/PipelineStageServiceImpl.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        if (value.isNullOrEmpty()) {
+            val pipelineIds = pipelineStageDao.getStagePipelineIdByProject(
+                dslContext = dslContext,
+                projectId = projectId,
+                startTime = startTime,
+                endTime = endTime,
+                tag = tag
+            )
+            redisOperation.set(
+                key = "MetricsOverviewStageDefaultPipelineIds:$projectId:$tag",
```
Comment (carlyin0801) [inline src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/PipelineStageServiceImpl.kt:137]: 这种直接缓存在内存中比较好，用caffeine

File: src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/constant/Constants.kt (hunk 1 above)
Comment (yjieliang) [reply src/backend/ci/core/metrics/api-metrics/src/main/kotlin/com/tencent/devops/metrics/constant/Constants.kt]: done

File: src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/PipelineStageServiceImpl.kt (hunk 3 above)
Comment (yjieliang) [reply src/backend/ci/core/metrics/biz-metrics/src/main/kotlin/com/tencent/devops/metrics/service/impl/PipelineStageServiceImpl.kt]: done

**Review by irwinsun — APPROVED** [review]
common-api里和Pipeline业务有关的Data Class 移到 api-process

File: src/backend/ci/core/process/biz-process/src/main/kotlin/com/tencent/devops/process/api/UserPipelineInfoResourceImpl.kt [hunk 4]
```diff
@@ -48,6 +50,24 @@ class UserPipelineInfoResourceImpl @Autowired constructor(
...
+        page: Int,
+        pageSize: Int
+    ): Result<Page<PipelineIdAndName>> {
+        checkParam(userId, projectId)
+        val result = pipelineListFacadeService.searchByProjectIdAndName(
+            projectId = projectId,
+            keyword = keyword,
+            page = page,
+            pageSize = pageSize,
+            channelCodes = listOf(ChannelCode.BS, ChannelCode.GIT)
```
Comment (irwinsun) [inline src/backend/ci/core/process/biz-process/src/main/kotlin/com/tencent/devops/process/api/UserPipelineInfoResourceImpl.kt:66]: 此处写死需要考虑后续优化，由用户侧传入

File: src/backend/ci/core/process/biz-base/src/main/kotlin/com/tencent/devops/process/engine/dao/PipelineInfoDao.kt [hunk 5]
```diff
@@ -200,6 +201,10 @@ class PipelineInfoDao {
                 query.and(CHANNEL.eq(channelCode.name))
             }
 
+            if (!keyword.isNullOrBlank()) {
+                query.and(PIPELINE_NAME.like("%$keyword%"))
```
Comment (irwinsun) [inline src/backend/ci/core/process/biz-base/src/main/kotlin/com/tencent/devops/process/engine/dao/PipelineInfoDao.kt:205]: 此处有模糊查询，需要确保ProjectId参数有效

**Review by irwinsun — APPROVED** [review]

**Pull request merged**